src/server/services/wms/qgswmsrenderer.cpp

```diff
@@ -739,12 +739,23 @@ namespace QgsWms
     ba = new QByteArray();
 
     QgsWmsParameters::Format infoFormat = mWmsParameters.infoFormat();
-    if ( infoFormat == QgsWmsParameters::Format::TEXT )
+    if ( infoFormat == QgsWmsParameters::Format::NONE )
+    {
+      throw QgsBadRequestException( QStringLiteral( "InvalidFormat" ),
```

Comment:
Will this leak `ba`?

Reply:
s/will this/this will/

unique_ptr ftw!

Reply:
QByteArray is implicitly shared, not sure there's an advantage in using pointers here anyway.

Reply:
Fixed after a rebase due to a modification https://github.com/qgis/QGIS/commit/25c27b3c80aeaca924140a66ae8e2ac4ce991f88 made by @jef-n :)